pkg/sdk/functions_gen_test.go

```diff
@@ -0,0 +1,528 @@
+						ColumnDataType: DataTypeVARCHAR,
+					},
+					{
+						ColumnName:     "country_name",
+						ColumnDataType: DataTypeVARCHAR,
+					},
+				},
+			},
+		}
+		returnNullValues := FunctionReturnNullValuesNotNull
+		opts.ReturnNullValues = &returnNullValues
+		nullInputBehavior := FunctionNullInputBehaviorCalledOnNullInput
+		opts.NullInputBehavior = &nullInputBehavior
+		returnResultsBehavior := FunctionReturnResultsBehaviorImmutable
+		opts.ReturnResultsBehavior = &returnResultsBehavior
+		opts.RuntimeVersion = String("2.0")
+		opts.Comment = String("comment")
+		opts.Imports = []FunctionImports{
+			{
+				Import: "@~/my_decrement_udf_package_dir/my_decrement_udf_jar.jar",
+			},
+		}
+		opts.Packages = []FunctionPackages{
+			{
+				Package: "com.snowflake:snowpark:1.2.0",
+			},
+		}
+		opts.Handler = String("TestFunc.echoVarchar")
+		opts.ExternalAccessIntegrations = []AccountObjectIdentifier{
+			NewAccountObjectIdentifier("ext_integration"),
+		}
+		opts.Secrets = []FunctionSecret{
+			{
+				SecretVariableName: "variable1",
+				SecretName:         "name1",
+			},
+			{
+				SecretVariableName: "variable2",
+				SecretName:         "name2",
+			},
+		}
+		opts.TargetPath = String("@~/testfunc.jar")
+		opts.FunctionDefinition = String("return id + name;")
+		assertOptsValidAndSQLEquals(t, opts, `CREATE OR REPLACE TEMPORARY SECURE FUNCTION IF NOT EXISTS %s (id NUMBER, name VARCHAR) COPY GRANTS RETURNS TABLE (country_code VARCHAR, country_name VARCHAR) NOT NULL LANGUAGE JAVA CALLED ON NULL INPUT IMMUTABLE RUNTIME_VERSION = '2.0' COMMENT = 'comment' IMPORTS = ('@~/my_decrement_udf_package_dir/my_decrement_udf_jar.jar') PACKAGES = ('com.snowflake:snowpark:1.2.0') HANDLER = 'TestFunc.echoVarchar' EXTERNAL_ACCESS_INTEGRATIONS = ("ext_integration") SECRETS = ('variable1' = name1, 'variable2' = name2) TARGET_PATH = '@~/testfunc.jar' AS 'return id + name;'`, id.FullyQualifiedName())
+	})
+}
+
+func TestFunctions_CreateFunctionForJavascript(t *testing.T) {
+	id := RandomSchemaObjectIdentifier()
+
+	defaultOpts := func() *CreateFunctionForJavascriptFunctionOptions {
+		return &CreateFunctionForJavascriptFunctionOptions{
+			name: id,
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		opts := (*CreateFunctionForJavascriptFunctionOptions)(nil)
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: incorrect identifier", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.name = NewSchemaObjectIdentifier("", "", "")
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("all options", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.OrReplace = Bool(true)
+		opts.Temporary = Bool(true)
+		opts.Secure = Bool(true)
+		opts.Arguments = []FunctionArgument{
+			{
+				ArgName:     "d",
+				ArgDataType: DataTypeFloat,
+			},
+		}
+		opts.CopyGrants = Bool(true)
+		float := DataTypeFloat
+		opts.Returns = &FunctionReturns{
+			ResultDataType: &float,
```

Comment:
can you please clarify? im not sure what you are asking for

Reply:
I was referring to DataTypeFloat being const instead of var (because you cannot get pointer to const) and here you're creating a temp value to take a pointer to it, so maybe we should define it as var instead of const.